common/src/stack/command/stack/commands/add/firmware/make/plugin_basic.py

```diff
@@ -0,0 +1,49 @@
+# @copyright@
+# Copyright (c) 2006 - 2018 Teradata
+# All rights reserved. Stacki(r) v5.x stacki.com
+# https://github.com/Teradata/stacki/blob/master/LICENSE.txt
+# @copyright@
+#
+# @rocks@
+# Copyright (c) 2000 - 2010 The Regents of the University of California
+# All rights reserved. Rocks(r) v5.4 www.rocksclusters.org
+# https://github.com/Teradata/stacki/blob/master/LICENSE-ROCKS.txt
+# @rocks@
+
+import stack.commands
+from stack.exception import ArgRequired, ArgError
+
+class Plugin(stack.commands.Plugin):
+	"""Attempts to add all provided make names to the database."""
+
+	def provides(self):
+		return 'basic'
+
+	def run(self, args):
+		# Require at least one make name
+		if not args:
+			raise ArgRequired(cmd = self.owner, arg = 'make')
+
+		# get rid of any duplicate names
+		makes = tuple(set(args))
+		# ensure the make name doesn't already exist
+		existing_makes = [
+			make
+			for make, count in (
+				(make, self.owner.db.count('(id) FROM firmware_make WHERE name=%s', make)) for make in makes
+			)
+			if count > 0
+		]
+		if existing_makes:
+			raise ArgError(cmd = self.owner, arg = 'make', msg = f'The following firmware makes already exist: {existing_makes}.')
+
+		for make in args:
+			self.owner.db.execute(
+				'''
+				INSERT INTO firmware_make (
+					name
+				)
+				VALUES (%s)
+				''',
+				make
+			)
```

Comment:
Minor syntax thing, no need to break the query onto multiple lines if it is short. I like the style:
```
self.owner.db.execute("""
	INSERT INTO firmware_make (name)
	VALUES (%s)
""", make)
```

OR, since I'm pretty sure it fits on a single line without wrapping (100 chars w/ tab==8 chars)
```
self.owner.db.execute('INSERT INTO firmware_make(name) VALUES (%s)', make)
```
Also, I believe the second parameter needs to be an iterable, unless mogrify is smart enough you are passing a single value. So:

```
self.owner.db.execute('INSERT INTO firmware_make(name) VALUES (%s)', (make,))
```

Reply:
I've been able to run these commands successfully passing a single param when I only have a single string format character embedded, so I think mogrify is smart enough?